Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -364,8 +364,31 @@ public static void Main(string[] args)
             File.Delete(fileName);
             Console.Out.WriteLine("Test4: PutobjectAsync with different content-type complete");
         }
-        private async static Task PutObject_Tester(MinioClient minio, string bucketName, string objectName, string fileName = null, string contentType = "application/octet-stream", long size = 0)
+
+        private async static Task PutObject_Test5(MinioClient minio)
+        {
+            Console.Out.WriteLine("Test5: PutobjectAsync with custom metadata");
+            string bucketName = GetRandomName(15);
+            string objectName = GetRandomName(10);
+            string fileName = CreateFile(1 * MB);
+            string contentType = "custom/contenttype";
+            Dictionary<string, string> metaData = new Dictionary<string, string>(){
+                { "x-amz-meta-customheader", "minio-dotnet"}
+            };
+            await Setup_Test(minio, bucketName);
+            ObjectStat statObject = await PutObject_Tester(minio, bucketName, objectName, fileName, contentType:contentType, metaData:metaData);
+            Assert.IsTrue(statObject != null);
+            Assert.IsTrue(statObject.metaData != null);
+            Assert.IsTrue(statObject.metaData.ContainsKey("x-amz-meta-customheader"));
```

Comment:
@poornas OK, I shall run the functional test and confirm. I wonder why you weren't able to see the assertion failure in your development setup. Do you know if header key comparison behaviour changed between dotnet versions?